### DISCUSSION TITLE
period output in drive cycle experiment

### DISCUSSION THREAD

**Discussion by dion-w** [pybamm-team/PyBaMM#1915] [Q&A] [answered]
Hey everyone, i encountered some other, for me, strange behaviour when using a drive cycle in an experiment. 

As it seems the period command doesn't work here. My simulation can do four driving cycles in the experiment, the simulation terminates during the fifth cycle at 2.5 V, but i don't get any data from this last cycle. Is there any way to get data until the actual voltage cut off?

I tried a number of formulations in the experiment but neither has the desired outcome:

"Run WLTC3 (W) (1 second period)" doesn't work as it has too many input arguments.
"Run WLTC3 (W) 1 second period" works but doesn't change anything, it seems to me that the period just gets ignored.
"Run WLTC3 (1 second period)" isn't possible as the unit for the drive cycle is needed.

Could you help me with this?

**Comment by dion-w** [comment, marked as answer]
Could anybody help? @rtimms @tinosulzer  I'd love to use this in a publication.

**Reply by dion-w** [reply]
So it doesn't seem to be a problem with period as "period" is calculated for the drive cycle automatically. It rather seems to be a problem with the stopping criterium where the experiment becomes "infeasible".

**Reply by rtimms** [reply]
Hi, yes this is probably related to #1902  . There is a bug in the way events are handled by the casadi solver which means it doesn't always catch events in drive cycles (sometimes the voltage drops below the cut-off but then increases again and the solver doesn't catch it). For now, a workaround is to use another solver (you will have to install one of the optional solvers like `scikit-odes` or `IDAKLU` if you are using a DAE model).
